scripts/db-heal-test.py

```diff
@@ -0,0 +1,231 @@
+    '--carmen-checkpoint-interval', str(checkpoint_granularity_flag), '--tracker-granularity',
+    str(checkpoint_granularity_flag), str(first_block), str(last_block)
+]
+
+os.chdir(aida_path)
+with open(aida_log_file, 'w') as f:
+    process = subprocess.Popen(cmd, stdout=f, stderr=subprocess.STDOUT)
+os.chdir(current_dir)
+
+print("Creating database with aida-vm-sdb...")
+
+# Start monitoring the log file
+latest_checkpoint = terminate_process_after(window, latest_checkpoint)
+
+# Wait for the first command to complete
+process.wait()
+
+if latest_checkpoint == -1:
+    sys.exit(1)
+
+# Find working directory
+working_dir = max(Path(tmp_path).iterdir(), key=os.path.getmtime)
+archive = working_dir / 'archive'
+live = working_dir / 'live'
+
+print("Testing db created, starting loop.")
+
+for i in range(1, number_of_iterations + 1):
+    last_working_dir = working_dir
+
+    # Find working dir - Aida copies db-src
+    working_dir = max(Path(tmp_path).iterdir(), key=os.path.getmtime)
+    archive = working_dir / 'archive'
+    live = working_dir / 'live'
+
+    # Dumb carmen's logs into a file to avoid spamming
+    c = open(carmen_log_file, 'w+')
+
+    # Restore Archive
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'reset', '--force-unlock', str(archive), str(latest_checkpoint)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
+
+    # Export genesis to restore LiveDB
+    genesis = Path(tmp_path) / 'test_genesis.dat'
+
+    print(f"Restoration complete. Exporting LiveDB genesis block {latest_checkpoint}.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'export', '--block', str(latest_checkpoint), str(archive), str(genesis)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
+
+    # Restore LiveDB
+    print("Export complete. Applying LiveDB genesis.")
+    try:
+        shutil.rmtree(live)
+        print("Live directory removed successfully.")
+    except FileNotFoundError:
+        print(f"Directory {live} does not exist.")
+        exit(1)
+    except PermissionError:
+        print(f"Permission denied to remove {live}.")
+        exit(1)
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        exit(1)
+
+    print("Importing LiveDB genesis.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'import-live-db', str(genesis), str(live)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
```

Comment:
nit: this block running a command and handling the error code is present several times; maybe it should be factory out into a utility function